subsys/bluetooth/mesh/Kconfig

```
@@ -1085,10 +1085,13 @@ menu "Firmware Update model configuration"
 
 config BT_MESH_DFU_FWID_MAXLEN
 	int "DFU FWID max length"
-	default 16
+	default 106
 	range 0 106
 	help
 	  This value defines the maximum length of an image's firmware ID.
+	  Please note that the only maximum length allows the device to be compliant with
+	  the Bluetooth Mesh DFU specification. If other than maximum length is set,
+	  then the device will fail with managing firmware images that have a longer FWID.
```

```suggestion
	  Please note that only the maximum length allows the device to be compliant with
	  the Bluetooth Mesh DFU specification. If other than maximum length is set,
	  then the device will fail with managing firmware images that have a longer FWID.
```

Is the "maximum length" referring to 106, or the value set by this config?

---

Also, "If other than maximum length is set" is a bit unclear to me, partly due to my above question. If it refers to 106 specifically I would suggest to specify this here.

---

I didn't get question. Could you elaborate?

---

Is it better now?

---

> I didn't get question. Could you elaborate?

My confusion was about

> If other than maximum value is set, then the device will fail with managing firmware images that have a longer FWID.

I'm not sure if this is a realistic situation but: I. e. if we set the range through the config to 86 (which now becomes the maximum), while the FWID length in practice is 70 (<86), does your comment warn about this specific case? Or is it in general if the FWID length is smaller than 106 (The maximum value for the maximum kconfig)?

---

I had the same confusion as Ingar, I see you updated it more clearly now!
However, it is not clear to me why only 106 allows the device to be compliant with the DFU spec. AFAIS the "Version Information" field is allowed (per Table 4.1) to be 0..106? Why have this option if we can only use the max?

---

To @HaavardRei, we add this and other such Kconfig options to allow users optimize their solutions as much as possible. If the use case is known in advance and users know that Vendor Information will never be greater than a certain value, it is fine to limit it. There is currently no tests in the test specification testing that your device can handle 106 bytes Vendor Information which makes your device unqualifiable. But even if there were such tests, users can still use stack for their use cases knowing the limitation.

---

Is there something that mandates this failure/incompatibility? Since this is Vendor Information (and could thus be whatever), why are we relying on this following an exact format? I guess this could be avoided if we utilized a length octet as the first octet of the field?

---

> If the Distributor was built with FWID 106, would it be able to distribute images to i. e. a bulb with length < 106 octets? If so, I think the description is a bit vague, and should specify that this recommendation/mandate is regarding the Distributor.

It is already:
```
If smaller than 106 octets are set, then the device will fail with managing firmware images that have a longer vendor-specific version information.
```
Updated device doesn't manage other images except its own.

Mentioning only distributor is not correct, since nobody prohibits user implementing initiator or combined roles like distributor + updated device or distributor + initiator.

---

> Is there something that mandates this failure/incompatibility? Since this is Vendor Information (and could thus be whatever), why are we relying on this following an exact format? I guess this could be avoided if we utilized a length octet as the first octet of the field?

The FW ID length is passed together with FW ID in every message, e.g. see FW Update Information Status message, section 5.1.2 (Current Firmware ID Length).

This Kconfig is primarily used in build asserts and runtime checks to ensure that Transport layer has enough buffers to process messages containing Firmware ID field.

> If the Distributor was built with FWID 106, would it be able to distribute images to i. e. a bulb with length < 106 octets?
If so, I think the description is a bit vague, and should specify that this recommendation/mandate is regarding the Distributor.

Yes, this Kconfig defines the longest possible length for Vendor Information field of Firmware ID state. But this is not only about Distributor, Initiator also has FW Update Client model. This Kconfig is used also in dfu_slot module which is used by Target role as well. This usage kind of prevents from storing Firmware ID longer than the configured value which aligns Target role behavior with Distributor built from the same code base (Zephyr Mesh).